R/qCount.R

```diff
@@ -3,516 +3,895 @@
+#' If \code{collapseBySample}=\code{TRUE}, groups of bam files with identical 
+#' sample name are combined by summing their alignment counts.
+#' 
+#' For \code{reportLevel="junction"}, the return value is a
+#' \code{GRanges} object. The start and end coordinates correspond to the
+#' first and last base in each detected intron. Plus- and minus-strand
+#' alignments are quantified separately, so that in an unstranded RNA-seq
+#' experiment, the same intron may be represented twice; once for each
+#' strand. The counts for each sample are contained in the \code{mcols}
+#' of the \code{GRanges} object.
+#' 
+#' @author Anita Lerch, Dimos Gaidatzis and Michael Stadler
+#' @keywords utilities misc
+#' 
+#' @export
+#' 
+#' @seealso 
+#' \code{\link[QuasR]{qAlign}},
+#' \code{\linkS4class{qProject}},
+#' \code{\link[parallel]{makeCluster}} from package \pkg{parallel}
+#' 
+#' @examples 
+#' library(GenomicRanges)
+#' library(Biostrings)
+#' library(Rsamtools)
+#' 
+#' # copy example data to current working directory
+#' file.copy(system.file(package="QuasR", "extdata"), ".", recursive=TRUE)
+#' 
+#' # load genome sequence
+#' genomeFile <- "extdata/hg19sub.fa"
+#' gseq <- readDNAStringSet(genomeFile)
+#' chrRegions <- GRanges(names(gseq), IRanges(start=1,width=width(gseq),names=names(gseq)))
+#' 
+#' # create alignments (paired-end experiment)
+#' sampleFile <- "extdata/samples_rna_paired.txt"
+#' proj <- qAlign(sampleFile, genomeFile, splicedAlignment=TRUE)
+#' 
+#' # count reads using a "GRanges" query
+#' qCount(proj, query=chrRegions)
+#' qCount(proj, query=chrRegions, useRead="first")
+#' 
+#' # hierarchical counting using a "GRangesList" query
+#' library(rtracklayer)
+#' annotationFile <- "extdata/hg19sub_annotation.gtf"
+#' gtfRegions <- import.gff(annotationFile, format="gtf", feature.type="exon")
+#' names(gtfRegions) <- mcols(gtfRegions)$source
+#' gtfRegionList <- split(gtfRegions, names(gtfRegions))
+#' names(gtfRegionList)
+#' 
+#' res3 <- qCount(proj, gtfRegionList)
+#' res3
+#' 
+#' # gene expression levels using a "TxDb" query
+#' library("GenomicFeatures")
+#' genomeRegion <- scanFaIndex(genomeFile)
+#' chrominfo <- data.frame(chrom=as.character(seqnames(genomeRegion)),
+#'                         length=end(genomeRegion),
+#'                         is_circular=rep(FALSE, length(genomeRegion)))
+#' txdb <- makeTxDbFromGFF(annotationFile, 
+#'                         format="gtf", 
+#'                         chrominfo=chrominfo,
+#'                         dataSource="Ensembl modified",
+#'                         organism="Homo sapiens")
+#'
+#' res4 <- qCount(proj, txdb, reportLevel="gene")
+#' res4
+#' 
+#' # exon-exon junctions
+#' res5 <- qCount(proj, NULL, reportLevel="junction")
+#' res5
+#' 
+#' @importFrom Rsamtools scanBamHeader
+#' @importFrom GenomeInfoDb seqlevels seqlevelsInUse seqlengths
+#' @importFrom parallel clusterMap clusterEvalQ splitIndices
+#' @importFrom GenomicRanges GRanges reduce findOverlaps seqnames
```

Comment:
The generic function for `seqnames` is also in `GenomeInfoDb`. I think we use only the `seqnames,GRanges-method` below. I am not sure what should be imported here. It seems that in other places we have been importing the generic rather than the specific method (e.g. `seqlevels` from `GenomeInfoDb` rather than from `BSgenome`).